src/Control/Effect/Parse.hs

```diff
@@ -27,3 +34,24 @@ parse :: (Member Parse sig, Carrier sig m)
       -> Blob
       -> m term
 parse parser blob = send (Parse parser blob pure)
+
+
+parseWith
+  :: (Carrier sig m, Member (Error SomeException) sig, Member Parse sig)
+  => [(Language, SomeParser c ann)]
+  -> (forall term . c term => term ann -> m a)
```

Comment:
Can you document what this is for (it’s an algebra, correct?)

Reply:
No, it’s just a function you run on an abstracted term under the constraint `c`.